posthog/temporal/batch_exports/batch_exports.py

```diff
@@ -672,15 +674,17 @@ async def execute_batch_export_insert_activity(
 
     except exceptions.ActivityError as e:
         if isinstance(e.cause, exceptions.CancelledError):
-            update_inputs.status = "Cancelled"
+            update_inputs.status = BatchExportRun.Status.CANCELLED
+        elif isinstance(e.cause, exceptions.ApplicationError) and e.cause.type not in non_retryable_error_types:
```

Comment:
@tomasfarias This was weird, but `non_retryable` [is a field on `ApplicationError`](https://github.com/temporalio/sdk-python/blob/1cfdaa86d476c238f54e3efc383d6cec8ccda123/temporalio/exceptions.py#L102-L106), but it doesn't seem to be correct here.

Meaning, `e.cause.type == Foo`, `non_retryable_error_types == ["Foo"]` and then `e.cause.non_retryable == False` (I would expect it to be `True`).

My check here does what I would expect (as tests prove), but I don't know if this is something we need to investigate more (is Temporal retrying non-retryable errors? Or is this just a weird SDK bug?). It seems unrelated to this PR, though.

Reply:
My guess is that application error doesn't make any assumptions about the underlying exception it's wrapping, so it sets `non_retryable` to the default of `False`.

Reply:
May be an SDK bug, may be they expect us to raise `ApplicationError` for this to work:

Did some digging, and I think we are landing here: https://github.com/temporalio/sdk-python/blob/f31f927b9e1604ebeccca02f4324658dfe741742/temporalio/converter.py#L796

Notice that if we are not one of Temporal's exception subclasses they are not checking whether we are `non_retryable`.

Reply:
Ah look at the docs, this is documented behavior: https://docs.temporal.io/references/failures#throw-from-activities

> During conversion, the following Application Failure fields are set:
> non_retryable is set to false.

Reply:
Good catch. That's so weird to me though, like are they actually respecting our retryable-ness?